src/arrayMethodReduce.js

```diff
@@ -5,7 +5,14 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let prev = arguments.length < 2 ? this[0] : initialValue;
+    const index = arguments.length < 2 ? 1 : 0;
```

Comment:
You could set prev and index in a single if block. But everything looks good.

Reply:
Thank you @ysrckr, I will consider your suggestion.